lib/Dialect/Arith/Conversions/ArithToModArith/ArithToModArith.cpp

```diff
@@ -160,6 +145,29 @@ struct GenericOpPattern : public OpConversionPattern<Op> {
   }
 };
 
+struct ConvertMemRefLoad : public OpConversionPattern<mlir::memref::LoadOp> {
+  ConvertMemRefLoad(mlir::MLIRContext *context)
+      : OpConversionPattern<mlir::memref::LoadOp>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      mlir::memref::LoadOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    // auto *parent = op.getMemRef().getDefiningOp();
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+
+    SmallVector<Type> retTypes;
+    if (failed(this->getTypeConverter()->convertTypes(op->getResultTypes(),
```

Comment:
again, this should use the adaptor which already has the type-converted results.

Reply:
Adaptors actually don't have the result types that are type converted. I forgot the reasoning behind this.

Reply:
Ah, thanks for the reminder!